### ISSUE TITLE
OPA gatekeeper - gatekeeper-update-namespace-label fails to pull image from private registry

### ISSUE THREAD

**Issue by izaac** [rancher/rancher#34000]
**Rancher Server Setup**
- Rancher version: `v2.6.0-rc1`
- Installation option: Single Airgap
   - Airgap, Downstream cluster is RKE1
- Proxy/Cert Details: rancher generated certs

**Information about the Cluster**
- Kubernetes version: `1.21.3`
- Cluster Type: 
   - If downstream, what type of cluster?: Custom RKE1 Airgap

**Describe the bug**

`ImagePullBackOff: Back-off pulling image "rancher/kubectl:v1.20.2"`

**To Reproduce**

- Setup an Airgap Rancher with Private Registry (basic auth)
- On an airgapped downstream cluster install OPA Gatekeeper from Dashboard

**Result**

`gatekeeper-update-namespace-label` fails to initialize the images.

**Expected Result**

Monitoring v2 installed

**Screenshots**

**Additional context**
`rancher-images.txt` includes `rancher/kubectl:v1.20.2`
The tagged image exist in the registry and is prefixed/tagged with the private registry host:

`{"name":"rancher/kubectl","tags":["v1.20.2","v1.18.0"]}`

**Comment by cbron** [comment]
Fix here is same as https://github.com/rancher/rancher/issues/33995
So only PR will be there, then this can go into test.

**Comment by jiaqiluo** [comment]
This issue is caused by the missing `{{ template "system_default_registry" . }}` in the following line
https://github.com/rancher/charts/blob/4ad408e0ca6f8c903c18553c32aa6540cf3b0b61/charts/rancher-gatekeeper/rancher-gatekeeper/100.0.0%2Bup3.5.1/templates/namespace-post-install.yaml#L33

**Comment by kinarashah** [comment]
Available to test with `v2.6.0-rc2`

**Comment by izaac** [comment]
Rancher version: `2.6.0-rc2`
Repo info:

```
spec:
  gitBranch: dev-v2.6
  gitRepo: https://git.rancher.io/charts
status:
  branch: dev-v2.6
  commit: abcc294eed065e8dec10a1757ed9dd5f808d15f4
```

The chart is now working on an Airgap HA setup with private registry.
